packages/textkit/src/engines/linebreaker/index.js

```diff
@@ -40,7 +41,12 @@ const breakLines = (string, nodes, breaks) => {
       end = prevNode.value.end;
 
       line = slice(start, end, string);
-      line = insertGlyph(line.length, HYPHEN, line);
+      // If the line is a URL Space character will be inserted (Inserting a hyphen will change the URL)
+      if (!line.isUrl) {
+        line = insertGlyph(line.length, HYPHEN, line);
+      } else {
+        line = insertGlyph(line.length, SPACE, line);
```

Comment:
Does this not work when a glyph is not inserted?

Reply:
No. It works. I'll remove this.